applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClientCredentials.java

```diff
@@ -0,0 +1,136 @@
+     */
+    public ConfigurationClientCredentials(String connectionString) {
+        credentials = new CredentialInformation(connectionString);
+        headerProvider = new AuthorizationHeaderProvider(credentials);
+    }
+
+    URL baseUri() { return this.credentials.baseUri; }
+
+    @Override
+    public Mono<String> authorizationHeaderValueAsync(HttpRequest request) {
+        try {
+            String authorizationValue = headerProvider.getAuthenticationHeaderValue(request);
+            return Mono.just(authorizationValue);
+        } catch (InvalidKeyException | NoSuchAlgorithmException e) {
+            return Mono.error(e);
+        }
+    }
+
+    private static class AuthorizationHeaderProvider {
+        private final String[] signedHeaders = new String[]{HOST_HEADER, DATE_HEADER, CONTENT_HASH_HEADER};
+        private final String signedHeadersValue = String.join(";", signedHeaders);
+        private final CredentialInformation credentials;
+
+        AuthorizationHeaderProvider(CredentialInformation credentials) {
+            this.credentials = credentials;
+        }
+
+        private String getAuthenticationHeaderValue(final HttpRequest request) throws NoSuchAlgorithmException, InvalidKeyException {
+            final String stringToSign = getStringToSign(request);
+            final Mac sha256HMAC = Mac.getInstance("HmacSHA256");
+            final SecretKeySpec secretKey = new SecretKeySpec(credentials.secret(), "HmacSHA256");
+
+            sha256HMAC.init(secretKey);
+
+            final String signature = Base64.getEncoder().encodeToString(sha256HMAC.doFinal(stringToSign.getBytes(StandardCharsets.UTF_8)));
+            return String.format("HMAC-SHA256 Credential=%s, SignedHeaders=%s, Signature=%s",
+                credentials.credential(),
+                signedHeadersValue,
+                signature);
+        }
+
+        private String getStringToSign(final HttpRequest request) {
+            String pathAndQuery = request.url().getPath();
+            if (request.url().getQuery() != null) {
+                pathAndQuery += '?' + request.url().getQuery();
+            }
+
+            final HttpHeaders httpHeaders = request.headers();
+            final String signed = Arrays.stream(signedHeaders)
+                .map(httpHeaders::value)
+                .collect(Collectors.joining(";"));
+
+            // String-To-Sign=HTTP_METHOD + '\n' + path_and_query + '\n' + signed_headers_values
+            // Signed headers: "host;x-ms-date;x-ms-content-sha256"
+            return String.format("%s\n%s\n%s", request.httpMethod().toString().toUpperCase(Locale.US), pathAndQuery, signed);
+        }
+    }
+
+    private class CredentialInformation {
+        private URL baseUri;
+        private String credential;
+        private byte[] secret;
+
+        URL baseUri() { return baseUri; }
+
+        String credential() { return credential; }
```

Comment:
Is there a reason we call value of `id` segment in connection string as credentials? is it really credential or just application id?

Reply:
Hmm.. not sure. It existed before. I'll update it to id. From the docs, it looks like just the service identity for it.